docs/references/go/verifying-sessions.mdx

```diff
@@ -3,100 +3,152 @@ title: Veryifing a session | Clerk Go SDK
 description: Learn how to verify a session with Clerk in your Go application.
 ---
 
-# Verifying a session in Go
+# Verifying a Clerk session in Go
 
-## Protect your backend APIs
+Go makes it really easy to create a simple HTTP server, and Clerk makes it simple to authenticate any request. The following examples show how to verify a session and retrieve the corresponding user.
 
-Go makes it really easy to create a simple HTTP server, and Clerk makes it simple to authenticate any request. The following examples shows how to verify a session and retrieve the corresponding user.
+## Using middleware
 
-<InjectKeys>
+The library provides two functions that can be used for adding authentication with Clerk to HTTP handlers.
 
-```go
-package main
+Both middleware functions support header based authentication with a bearer token. The token is parsed, verified and
+its claims are extracted as [SessionClaims](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaims).
+
+The claims will then be made available in the `http.Request.Context` for the next handler in the chain. The library
+provides a helper for accessing the claims from the context, [SessionClaimsFromContext](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaimsFromContext).
 
+The two middleware functions are [WithHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#WithHeaderAuthorization)
+and [RequireHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#RequireHeaderAuthorization).
+Their difference is that the `RequireHeaderAuthorization` calls `WithHeaderAuthorization` under the hood, but responds
+with HTTP 403 Forbidden if it fails to detect valid session claims.
+
+Your Clerk secret key is required. If you are signed into your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.
```

Comment:
```suggestion
<Callout type="info">

Your Clerk secret key is required. If you are signed into your Clerk Dashboard, your secret key should become visible by selecting the eye icon below. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.

</Callout>
```

Reply:
Done!